lib/sms/sms_submit.c

```diff
@@ -88,6 +104,80 @@ static int sms_submit_encode_number(
 	return 0;
 }
 
+static void sms_submit_encode_sca(char *sca_field)
+{
+	int ret;
+
+	uint8_t encoded_sca[SMS_MAX_ADDRESS_LEN_CHARS + 1];
+	uint8_t encoded_sca_size;
+	uint8_t encoded_sca_size_octets = 0;
+	/* 36 is enough and has few characters extra. Maximum required should be:
```

Comment:
The size does not match the implementation anymore, you could consider removing the size of the buffer from the comment.